src/Swarrot/Processor/XDeath/README.md

```diff
@@ -0,0 +1,103 @@
+# XDeathMaxCountProcessor
+
+XDeathMaxCountProcessor is a [swarrot](https://github.com/swarrot/swarrot) processor.
+Its goal is to execute a callback when x-death max count reached in order to kill poison message.
+
+This is a very different approach than RetryProcessor because there is no software clone message republish.
+This processor use the native rabbitmq x-death functionnality.
+
+## Configuration
+
+|Key                                  |Default|Description                                                                  |
+|:-----------------------------------:|:-----:|-----------------------------------------------------------------------------|
+|x_death_max_count                    |300    |The number of attempts before callback.
+|x_death_max_count_log_levels_map     |[]     |Map of classes to a log level when retry. (Warning by default)               |
+|x_death_max_count_fail_log_levels_map|[]     |Map of classes to a log level when all retries failed. (Warning by default)  |
+
+## How it works
+
+When your processor throw an exception (= failed to process a message), the
+XDeathMaxCount Processor will catch it and execute the callback if the message header x-death count
+exceed the `x_death_max_count`.
+
+/!\ Caution ! While your max count callback return null the exception gonna to be rethrow
+
+## Real example
+
+Let's say you want to consume a queue named `mail`. You would like to have 3
+retries if the mail is not sent correctly. Each retry was separate by 30 seconds.
+
+First let's create the delaying exchange+queue 
+
+* Create an exchange `waiting_30` (type: `topic`)
+* Create an queue `waiting_30` (x-dead-letter-exchange: ``, x-message-ttl: `30000`)
```

Comment:
The 2 backticks look weird here. They are not creating an inline docblock in the markdown rendering. What are you trying to represent here ?

Reply:
an empty string (to bind the queue to the AMQP Default exchange)

Reply:
For things to work, you need to put a space between the start and the end delimiter of the code span (the content between delimiter is trimmed, so you will get an empty string in the end, but it is needed to separate them to create a code span)